### PR TITLE
Make mechanism of moving users to the beta feature, separate between games saved on frontend, games saved on backend and games shared

### LINKED ISSUE #2425: Separate between rooms using the new saving feature, rooms that have been shared and rooms that are on the old saving feature
This separation should be done as soon as one of the new features is activated, and would be represented by a bool in the db. They should also have some kind of icons in the my games section and rooms using the new saving feature should not be able to use the old one also, because of how the saving server works.

### PR DESCRIPTION
**Pull request by Cosmin-Mare**
Make a check in /~/[id] page that renders a different page depending on whether the user should be in the beta testing group.
The emails that are hardcoded to be moved to the new feature are [development@hackclub.com](mailto:development@hackclub.com), [cosmin@hackclub.com](mailto:cosmin@hackclub.com) and [graham@hackclub.com](mailto:graham@hackclub.com)
fixes https://github.com/hackclub/sprig/issues/2412/

Adds icons to games that have been shared/saved in the backend. A game that was saved on the backend can't be moved to the frontend saving strat. When a game upgrades to the backend saving(for example the users goes to /beta/ or the user gets upgraded to a beta user), the game is duplicated and a new copy which is now using the backend saving strat is opened. When a room is shared it doesn't make a copy of it before(should it?).

Reminder for self to update the saving server when this gets merged to only save games that have the isSavedOnBackend active.

fixes #2425

popups for indicating the saving and collab state of games on my games page
image
image

### CODE REVIEW

**Review by grymmy — COMMENTED** [review]
Note that this PR includes code that I've already given feedback on in #2414

**Comment by grymmy** [comment]
Also needs to have conflict resolved.

File: src/lib/game-saving/account.ts [hunk 1]
```diff
@@ -331,4 +336,27 @@ export const getSnapshotData = async (id: string): Promise<SnapshotData | null>
...
+
+async function hashCodeToBigInt(string : string) : Promise<bigint>{
+	return BigInt(`0x${ await sha256Hash(string)}`);
+}
+
+export async function isAccountWhitelistedToUseCollabAndSavingBetaFeatures(id: string, email: string) : Promise<boolean>{
+	if(import.meta.env.PERCENT_OF_USERS_WHITELISTED_FOR_BETA_FEATURE == 0 || import.meta.env.PERCENT_OF_USERS_WHITELISTED_FOR_BETA_FEATURE == undefined) return false;
+	let hashedId = await hashCodeToBigInt(id);
+	
+	if(hashedId % BigInt(100) < import.meta.env.PERCENT_OF_USERS_WHITELISTED_FOR_BETA_FEATURE ||
```
Comment (grymmy) [inline src/lib/game-saving/account.ts:353]: Good that you modified this to be percentage based, so much easier to understand now.

**Review by grymmy — APPROVED** [review]
lgtm

File: src/lib/game-saving/account.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Cosmin-Mare) [reply src/lib/game-saving/account.ts:353]: iirc that's the syntax, you can't import like you import other classes. the most one can do is move it to a const at the top of the file, but that's not done anywhere in the project:))

**Pull request merged**